src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py

```diff
@@ -156,47 +60,88 @@ def __init__(
                     component_category={ComponentCategory.EV_CHARGER}
                 )
             }
-        self._channel = Broadcast[EVChargerPoolStates](
-            "EVCharger States", resend_latest=True
+        self._namespace = f"ev-charger-pool-{uuid.uuid4()}"
+        self._formula_pool = FormulaEnginePool(
+            self._namespace,
+            self._channel_registry,
+            self._resampler_subscription_sender,
         )
-        self._task: Optional[asyncio.Task[None]] = None
-        self._merged_stream: Optional[Merge] = None
-
-    async def _run(self) -> None:
-        logger.debug("Starting EVChargerPool for components: %s", self._component_ids)
-        api_client = microgrid.get().api_client
-        streams: list[Receiver[EVChargerData]] = await asyncio.gather(
-            *[api_client.ev_charger_data(cid) for cid in self._component_ids]
+
+    async def total_current(self) -> FormulaReceiver3Phase:
+        """Fetch the total current for the ev chargers in the pool.
+
+        If a formula engine to calculate ev charger current is not already
+        running, it will be started.  Else, we'll just get a new receiver to the
+        already existing data stream.
```

Comment:
```suggestion
        If a formula engine to calculate EV charger current is not already
        running, it will be started. Else, it'll just get a new receiver to the
        already existing data stream.
```

Reply:
please discard the double-space suggestion but the other ones still might apply

Reply:
reworded the "It'll get a" to "It'll return a" in a separate commit.